docs/reference-implementation/Developer-Guide.md

```diff
@@ -0,0 +1,58 @@
+# Developer Guide
+
+## Tools Required
+
+The following tools and applications should be installed on the client machine used to effectively develop or deploy the Azure Mission-Critical reference implementation:
+
+- Install [Azure CLI](https://docs.microsoft.com/cli/azure/service-page/azure%20cli?view=azure-cli-latest)
+
+- Install [Azure DevOps CLI](https://docs.microsoft.com/azure/devops/cli/?view=azure-devops)
+
+- Install [PowerShell](https://docs.microsoft.com/powershell/scripting/install/installing-powershell-on-windows?view=powershell-7.1).
+
+- Install [Visual Studio Code](https://code.visualstudio.com/Download)
+
+- Install [Terraform CLI](https://www.terraform.io/downloads)
+
+## Structure of the Azure Mission-Critical reference implementation
+
+All infrastructure code lives in the [workload folder](/src/infra/workload/), where they are separated by [global](/src/infra/workload/globalresources/) and [releaseunit](/src/infra/workload/releaseunit/). Each service has a .tf file (Terraform Template) that contains its definition and configuration. If you need to make a change to the definition or configuration of a service, you will need to locate its .tf file. 
+
+
+### Example
+
+You want to turn off zone redundancy for Cosmos DB. Go to [cosmosdb.tf](/src/infra/workload/globalresources/cosmosdb.tf) and change the `zone_redundant` field to false under `geo-location` 
+
+`dynamic "geo_location" {
+    for_each = var.stamps
+    content {
+      location          = geo_location.value
+      failover_priority = geo_location.key
+      zone_redundant    = true --> **false**
+    }
+  }`
+
+  If you want to make an addition to a service such as adding an additional database to Cosmos DB, you will need to locate all files that interact with the service.
+  
+  ### Example
+
+  You want to add another database to Cosmos DB for Inventory. You need to instantiate another database in the cosmosdb.tf and edit all the files that contain variables or variable references for database names.
+
+  1. Go to the [global varibles file](/src/infra/workload/globalresources/variables.tf). Copy and paste lines 67-71. Replace `cosmosdb_database_name` with a new name for your database (e.g. `cosmosdb_inventory_database_name`).
```

Comment:
Maybe try not to reference exact line numbers, since these can change all the time and nobody will remember to update this guide